modules/edu.gemini.seqexec.web/edu.gemini.seqexec.web.client/src/main/scala/edu/gemini/seqexec/web/client/semanticui/elements/icon/Icon.scala

```diff
@@ -0,0 +1,73 @@
+package edu.gemini.seqexec.web.client.semanticui.elements.icon
+
+import japgolly.scalajs.react.vdom.prefix_<^._
+import japgolly.scalajs.react.{ReactComponentB, ReactNode}
+
+/**
+  * Semantic UI Icon component
+  */
+object Icon {
+  sealed trait Flipped
+  case object NotFlipped extends Flipped
+  case object Horizontally extends Flipped
+  case object Vertically extends Flipped
+
+  sealed trait Rotated
+  case object NotRotated extends Rotated
+  case object Clockwise extends Rotated
+  case object CounterClockwise extends Rotated
+
+  case class Props(id: String,
+                   disabled: Boolean = false,
+                   loading: Boolean = false,
+                   fitted: Boolean = false,
+                   tiny: Boolean = false,
+                   mini: Boolean = false,
+                   small: Boolean = false,
+                   large: Boolean = false,
+                   big: Boolean = false,
+                   huge: Boolean = false,
+                   massive: Boolean = false,
+                   link: Boolean = false,
+                   flipped: Flipped = NotFlipped,
+                   rotated: Rotated = NotRotated,
+                   circular: Boolean = false,
+                   bordered: Boolean = false,
+                   inverted: Boolean = false,
+                   color: Option[String] = None)
```

Comment:
I take it these are attributes on some given UI component that we're giving a type?  Since the model supports it, what would it mean to be tiny, but not mini, but small, but not large, but huge but not massive?  Should this shunted into case objects like you did with `Flipped` and `Rotated`?